### ISSUE TITLE
[BUG] - Using Authelia with OIDC comes back to login page

### ISSUE THREAD

**Issue by shomilsaini** [mealie-recipes/mealie#3671]
### First Check

- [X] This is not a feature request.
- [X] I added a very descriptive title to this issue (title field is above this).
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.
- [ ] This issue can be replicated on the demo site (https://demo.mealie.io/).

### What is the issue you are experiencing?

I Setup Authelia with Mealia as per docs
https://www.authelia.com/integration/openid-connect/mealie/

### Steps to Reproduce

go to mealia.xyz.com click "Login with Oauth"
verified by Authelia using 2FA page redirects to Mealie login page. Does not login

### Please provide relevant logs

INFO     2024-05-31T16:21:01 - [xxx.xxx.xxx.xxx:0] 200 OK "GET /api/app/about HTTP/1.1"
INFO     2024-05-31T16:21:01 - [xxx.xxx.xxx.xxx:0] 200 OK "GET /api/app/about HTTP/1.1"
INFO     2024-05-31T16:21:01 - [xxx.xxx.xxx.xxx:0] 200 OK "GET /api/app/about HTTP/1.1"
INFO     2024-05-31T16:21:01 - [xxx.xxx.xxx.xxx:0] 200 OK "GET /api/app/about/startup-info HTTP/1.1"
INFO     2024-05-31T16:21:18 - [xxx.xxx.xxx.xxx:0] 307 Temporary Redirect "GET /login?code=authelia_ac_TCXyvJWKMc4PYleH502aMvXUUZ6TaMnozkG8CoB9etY.0UevViB5EBib9mXfxBNdOQiJ9EeLbQTfZnPPPb0zsWY&iss=https%3A%2F%2Fauth.xyz.com&scope=openid+profile+email+groups&state=G7dmMCcaqQ HTTP/1.1"
INFO     2024-05-31T16:21:18 - [xxx.xxx.xxx.xxx:0] 200 OK "GET /login/?code=authelia_ac_TCXyvJWKMc4PYleH502aMvXUUZ6TaMnozkG8CoB9etY.0UevViB5EBib9mXfxBNdOQiJ9EeLbQTfZnPPPb0zsWY&iss=https%3A%2F%2Fauth.xyz.com&scope=openid+profile+email+groups&state=G7dmMCcaqQ HTTP/1.1"
INFO     2024-05-31T16:21:19 - [xxx.xxx.xxx.xxx:0] 200 OK "GET /api/app/about HTTP/1.1"
INFO     2024-05-31T16:21:19 - [xxx.xxx.xxx.xxx:0] 200 OK "GET /api/app/about/startup-info HTTP/1.1"
INFO     2024-05-31T16:21:19 - [xxx.xxx.xxx.xxx:0] 200 OK "GET /api/app/about HTTP/1.1"

### Mealie Version

Version
v1.7.0

Build
[d1c5a6ed8c28e6635af4ac189f878fbfbada2236](https://github.com/mealie-recipes/mealie/commit/d1c5a6ed8c28e6635af4ac189f878fbfbada2236)

### Deployment

Unraid

**Comment by shomilsaini** [comment]
INFO     2024-05-31T16:33:55 - [xxx.xxx.xxx.xxx:0] 200 OK "GET /api/app/about HTTP/1.1"
INFO     2024-05-31T16:33:55 - [xxx.xxx.xxx.xxx:0] 200 OK "GET /api/app/about HTTP/1.1"
INFO     2024-05-31T16:33:55 - [xxx.xxx.xxx.xxx:0] 200 OK "GET /api/app/about/startup-info HTTP/1.1"
INFO     2024-05-31T16:33:55 - [xxx.xxx.xxx.xxx:0] 200 OK "GET /api/app/about HTTP/1.1"
INFO     2024-05-31T16:34:06 - [xxx.xxx.xxx.xxx:0] 307 Temporary Redirect "GET /login?code=authelia_ac_AS8WtrJr8zjc1ijzM_Lk3UHoEcE7WGvu4tzkQYFimew.kInGdNIHh-vaSa9DIGURVV8Goh4qV_idoJkk1kyL7no&iss=https%3A%2F%2Fauth.xyz.com&scope=openid+profile+email+groups&state=o1VloAl8ov HTTP/1.1"
INFO     2024-05-31T16:34:06 - [xxx.xxx.xxx.xxx:0] 200 OK "GET /login/?code=authelia_ac_AS8WtrJr8zjc1ijzM_Lk3UHoEcE7WGvu4tzkQYFimew.kInGdNIHh-vaSa9DIGURVV8Goh4qV_idoJkk1kyL7no&iss=https%3A%2F%2Fauth.xyz.com&scope=openid+profile+email+groups&state=o1VloAl8ov HTTP/1.1"
INFO     2024-05-31T16:34:06 - [xxx.xxx.xxx.xxx:0] 200 OK "GET /api/app/about HTTP/1.1"
INFO     2024-05-31T16:34:06 - [xxx.xxx.xxx.xxx:0] 200 OK "GET /api/app/about/startup-info HTTP/1.1"
INFO     2024-05-31T16:34:06 - [xxx.xxx.xxx.xxx:0] 200 OK "GET /api/app/about HTTP/1.1"

**Comment by shomilsaini** [comment]
Works now,
if anyone wondering add OIDC_USER_CLAIM [Docs](https://docs.mealie.io/documentation/getting-started/installation/backend-config/#openid-connect-oidc)

Also refer to: [GitHub](https://github.com/mealie-recipes/mealie/issues/3473#issuecomment-2060508808)

**Comment by shomilsaini** [comment]
I tried this variable [OIDC_TLS_CACERTFILE](https://docs.mealie.io/documentation/getting-started/installation/backend-config/#openid-connect-oidc) but that always gives error. Not clear documentation on what cert to provide. But it works without it on cloudflare Origin Cert on my authelia.

Thanks!

**Comment by Diaoul** [comment]
Had the same issue on a well configured Authelia
Purging cookies and spamming the login button worked

**Comment by rdgerken** [comment]
I'm still having this exact problem - even after adding   OIDC_USER_CLAIM: email   to my docker environment variables. Did you have something different?  I also tried OIDC_USER_CLAIM: preferred_username

I've run out of things to try.

(Running Cloudflare/Traefik/Authelia here)

**Comment by MRobi1** [comment]
> I'm still having this exact problem - even after adding OIDC_USER_CLAIM: email to my docker environment variables. Did you have something different? I also tried OIDC_USER_CLAIM: preferred_username
> 
> I've run out of things to try.
> 
> (Running Cloudflare/Traefik/Authelia here)

Running up against the same thing here but with Authentik.
V1.9.0

I've noticed after authenticating, I'm redirected to this URL
https://mealie.mydomain.xyz/login/?code=longstringhere&state=shortstringhere

cleared cache in cloudflare and accessing through incognito.

**Comment by rdgerken** [comment]
I'm about to give up. I looked at all closed issues regarding this OIDC redirect back to login thing - and it's still happening. I get successful messages on the Authelia side (completely bypassing Authelia forward auth).

I go to the Mealie login page - select login with OIDC button
takes me to Authelia - I perform my login.
once accepted, Authelia presents me with the consent page - I accept
boom - right back to Mealie login page.

On the Mealie side, I have a 307 redirect with /login/?code=longstring&scope=openid+profile+emails+groups&state=shortcode

that is immediately followed with a 200 OK GET with the same data as the 307
that is followed with 2 more 200 OK GET /api/app/about
and then I get a 304 Not Modified "GET /sw.js"  - I don't know if this is expected or not?

On the Authelia log, everything looks normal. I see an auth request being processed, and an auth request  success message.

and that's it... 

I've tried 1.7, 1.9, and nightly. I've tried several variations of configurations on both sides, all with the same outcome. Also bypassing cloudflare caching and tried also with private browser windows.

**Comment by cmintey** [comment]
Have you enabled debug logging? There are some helpful debug log messages for OIDC that can help track down the issue on why you're not authenticating with Mealie. Set `LOG_LEVEL=DEBUG` in your environment variables. Please share logs as well, both the server debug logs and any browser console logs

**Comment by rdgerken** [comment]
@cmintey - thanks for pushing me to look closer at my logs. I went off on a tangent and got Loki set up, and then once in the browser logs, I found that I had a CORS header issue. I was able to reconfigure Traefik to provide the proper CORS headers, and then I had a few other problems to resolve - (such as there doesn't seem to be a way to specify what mealie group an OIDC created user should be created in (and I changed my group name from "Home" to something else), and then also a header buffer size issue with Authelia). Once I got those items dialed in, everything is working as it should. -Thanks again for being willing to help!
